File: 251214/기수 정렬 구현/implement-radix-sort.js


const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
let arr = input[1].split(' ').map(Number);

// Please write your code here.

// k 구하기
let max = Math.max(...arr)
let k = 0

while (max >= 1) {
    k++
    max /= 10
}

// sort
for(let target = 0; target < k; target++) {
    // radixArr setting
    const radixArr = []
    for(let i = 0; i < 10; i++)
        radixArr.push([])

    // sort by ith number
    for(let i = 0; i < n; i++) {
        
        let ith = Math.floor(arr[i] / Math.pow(10, target)) % 10
        radixArr[ith].push(arr[i])
    }

    // make radixArr to 1 demension arr
    let ret = []
    for(let i = 0; i < 10; i++) {
        for(let j = 0; j < radixArr[i].length; j++) {
            ret.push(radixArr[i][j])
        }
    }

    arr = ret
}

for (let i = 0; i < n; i++) {
    process.stdout.write(arr[i]+" ")
}

File: 251214/기수 정렬 구현/implement-radix-sort.py
import math

n = int(input())
arr = list(map(int, input().split()))

mx = 0;
for i in range(n):
    if mx < arr[i]:
        mx = arr[i]

k = 0
while(mx >= 1):
    k += 1
    mx /= 10

for i in range(k):
    t_a = [[] for row in range(10)] 
    for j in range(n):
        x = math.floor(arr[j] / (10 ** (k - i - 1)) % 10)
        t_a[x].append(arr[j])
    r_a = []
    for j in range(10):
        for k in range(len(t_a[j])):
            r_a.append(t_a[j][k])

    arr = r_a

for i in range(n):
    print(arr[i], end=" ")
